cmd/mrregister.cpp

```diff
@@ -107,24 +118,49 @@ void usage ()
 
   + Registration::fod_options
 
+  + multiContrastOptions
+
   + DataType::options();
 }
 
 using value_type = double;
 
+void run () {
 
+  vector<Header> input1, input2;
+  const size_t n_images = 1 + (argument.size() - 2 ) / 3;
+  { // parse arguments and load input headers
+    if (MR::abs(float (n_images) - (1.0 + float (argument.size() - 2 ) / 3.0)) > 1.e-6) {
```

Comment:
This would make more sense as a modulo operation

Reply:
I can change line 133 to `if (n_images - 1) * 3 + 2 != argument.size()` if you don't like the floating-point operations but not sure how to use the modulo to make it more intuitive? The arguments are paired input images (subject i and j), with contrast pairs separated by '+': 
For instance: `['i1', 'j1', '+', 'i2', 'j2', '+', 'i3', 'j3']`.

Reply:
For the case of SH data, the logic that makes most sense to me is something like what's used in [`Math::SH::check()`](https://github.com/MRtrix3/mrtrix3/blob/dev/core/math/SH.h#L696-L698): Map number of volumes to *lmax*, then check that *lmax* maps to the number of volumes. We could put such functions in a module if it'd be useful. What you've got here is the same functionality, just not functionalised.

What I was thinking in terms of a modulo operation basically avoids having to derive the inverse function: take the expression for `n_images`, replace the division with a modulo, and make sure it's zero.

Reply:
I think you refer to the number of volumes check. That's basically how I implemented it [link](https://github.com/MRtrix3/mrtrix3/pull/1798/commits/d854e3320a2450c22c7b38f1b1f139671ad94405#diff-2a6687b787e993424fe0d35053e28d72R251).

Reply:
I was thinking since that logic appears within `population_template`, Python versions of those functions might be handy.